### PR TITLE
deps: bump react-hook-form to 7.X

### PR DESCRIPTION
**Pull request by romainseb**
**What is the problem this PR is trying to solve?**
react-hook-form is used in version 6 and version 7 is out for while now

**What is the chosen solution to this problem?**
Upgrade it.

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: .changeset/lovely-chefs-remain.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@talend/design-docs': minor
+'@talend/react-stepper': minor
+'@talend/react-forms': minor
```
Comment (romainseb) [inline .changeset/lovely-chefs-remain.md:4]: Looks not clear for me if it's a breaking change or not.
(i think it is if we load RHF as a UMD in the apps)

**Comment by jmfrancois** [comment]
image

File: .changeset/lovely-chefs-remain.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+'@talend/design-docs': major
+'@talend/react-stepper': major
+'@talend/react-forms': major
+---
+
+deps: bump react-hook-form to 7.X
```
Comment (jmfrancois) [inline .changeset/lovely-chefs-remain.md:7]: We should explain the breaking change here
```suggestion
deps: bump react-hook-form to 7.X

So if you are using rhf in your project you should also bump it and apply the migration guide
https://legacy.react-hook-form.com/migrate-v6-to-v7/
```

**Review by jmfrancois — APPROVED** [review]
just take a look at the proposal on the changelog. For the rest :+1:

**Review by smouillour — APPROVED** [review]

**Review by tlnd-mhuchet — APPROVED** [review]

**Pull request merged**